### PR TITLE
Fix AddExchanges filter predicate when filter pushhdown is enabled

### PR DESCRIPTION
**Pull request by sachdevs**
```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/AddExchanges.java [hunk 1]
```diff
@@ -557,7 +557,7 @@ public PlanWithProperties visitDistinctLimit(DistinctLimitNode node, PreferredPr
         @Override
         public PlanWithProperties visitFilter(FilterNode node, PreferredProperties preferredProperties)
         {
-            if (node.getSource() instanceof TableScanNode) {
+            if (node.getSource() instanceof TableScanNode && metadata.isLegacyGetLayoutSupported(session, ((TableScanNode) node.getSource()).getTable())) {
```
Comment (highker) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/AddExchanges.java:560]: Add a comment to explain the logic in details.

Comment (sachdevs) [reply presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/AddExchanges.java:560]: Added comment + test

**Comment by sachdevs** [comment]
No regressions in last test run: https://travis-ci.org/github/prestodb/presto/builds/672166804. Added tests and comments.

**Comment by highker** [comment]
@sachdevs, as @mbasmanova has mentioned, could you validate the test will fail without this patch? If that is the case, I will merge the PR.

**Comment by sachdevs** [comment]
Yup I did check that when I added it. It fails without the patch.

**Pull request merged**